EchoPro/survey.py

```diff
@@ -327,269 +328,502 @@ def read_validated_data( self ,
         else:
             raise ValueError('Unexpected data attribute structure. Check API settings located in the configuration YAML and core.py')
         
-        # ------------------------------------------------------------------------------------------------
-        # THIS IS A TEMPORARY FUNCTION
-        # -- This PARASITIC_TREE function is temporarily here to continue validating data attributes are
-        # -- being organized in an expected manner
-        # self.PARASITIC_TREE( config_map )
-        
     def biometric_distributions( self ):
         """
         Expand bin parameters into actual bins for length and age distributions
         """
 
         # Pull the relevant age and length bins and output a dictionary
-        biometrics = {
-            'length_bins_arr': np.linspace( self.config['bio_hake_len_bin'][0] ,
-                                        self.config['bio_hake_len_bin'][1] ,
-                                        self.config['bio_hake_len_bin'][2] ,
-                                        dtype = np.int64 ) ,
-            'age_bins_arr': np.linspace( self.config['bio_hake_age_bin'][0] ,
-                                    self.config['bio_hake_age_bin'][1] , 
-                                    self.config['bio_hake_age_bin'][2] ) ,
-        }
-
-        # Update the configuration so these variables are mappable
+        length_bins = np.linspace( self.config['bio_hake_len_bin'][0] ,
+                           self.config['bio_hake_len_bin'][1] ,
+                           self.config['bio_hake_len_bin'][2] ,
+                           dtype = np.float64 )
+
+        age_bins = np.linspace( self.config['bio_hake_age_bin'][0] ,
+                                self.config['bio_hake_age_bin'][1] , 
+                                self.config['bio_hake_age_bin'][2] ,
+                                dtype = np.float64 )
+
+        ### Discretize parameters into arrays of binned values that will be used to 
+        ### quantize the distributions in calculations downstream
+        # Determine bin widths
+        length_binwidth = np.mean( np.diff( length_bins / 2.0 ) )
+        age_binwidth = np.mean( np.diff( age_bins / 2.0 ) )
+
+        # Now the bins are centered with the first and last elements properly appended
+        # These create an along-array interval such that values can be cut/discretized into
+        # bins that fall between each value/element of the array
+        length_centered_bins = np.concatenate( ( [ length_bins[0] - length_binwidth ] ,
+                                                length_bins + length_binwidth ) )
+        age_centered_bins = np.concatenate( ( [ age_bins[0] - age_binwidth ] ,
+                                            age_bins + age_binwidth ) ) 
+
+        # Construct dictionary that will be move values within the configuration file 
```

Comment:
Some words are missing or there's a typo of some kind in this sentence?

Reply:
Maybe as a result, I am not understanding why the same parameters are saved again in a different place in `self.config`. Perhaps `self.config['bio_hake_len_bin']` should just be saved in `self.config['biometrics']['bio_hake_len_bin']` when it's first parsed and saved?